### ISSUE TITLE
id is null in getter of transient property

### ISSUE THREAD

**Issue by andreasschiestl** [grails/grails-core#9955]
Hi!

I'm not sure if it is a Grails Bug. I have a abstract base class (Customer) and two derived classes (PrivateCustomer and BusinessCustomer). 

This is a simplyfied version of the Customer domain: 
```
abstract class Customer {

    String info

    static constraints = {
        info (nullable: true)
    }

    static mapping = {
        tablePerHierarchy false
        table ("bs_core_customer")

        info (type: "text")
    }

    static transients = ["displayId", "displayName"]

    String getDisplayId() {
        return String.format("%05d", id)
    }

    abstract String getDisplayName()
}
```

The issue is that the value of id is null inside of getDisplayId, BUT only for BusinessCustomer objects, not for PrivateCustomer objects. 

Another strange thing is, if I try to inspect the id, IntelliJ IDEA shows the correct value:
![bildschirmfoto 2016-05-23 um 22 20 34](https://cloud.githubusercontent.com/assets/8170218/15483372/a2d7c6ea-2134-11e6-8b37-318ce635aea0.png)

I have tried to create a test project, but unfortunately I cannot reproduce the error.

### Environment Information

- **Operating System**: Mac OS X 10.11.4 (15E65)
- **Grails Version:** 3.1.7
- **JDK Version:** 1.8.0_72
- **Groovy Version:** 2.4.6

**Comment by KerchumA222** [comment]
You should also post your other 2 domains (the ones that inherit from Customer).

**Comment by andreasschiestl** [comment]
This is the BusinessCustomer domain:
```
class BusinessCustomer extends Customer {

    String name
    String vatIdentificationNumber

    static hasMany = [
            contactPersons: ContactPerson,
    ]

    static constraints = {
        vatIdentificationNumber (nullable: true)
    }

    static mapping = {
        table ("bs_core_businesscustomer")
    }

    @Override
    String getDisplayName() {
        return name
    }
```

The PrivateCustomer is nearly the same, except other properties and some oder 1:n relation. But nothing that interact with getDisplayId.

**Comment by graemerocher** [comment]
Please upload an example that reproduces the issue.

**Comment by andreasschiestl** [comment]
I have already tried to create a small test project that reproduces the issue, but unfortunately in the test project the issue does not appear and the original project is a commercial projekt which I cannot share on github.

But I did find something interesting. If I add id to the domain properties, everything works as expected
```
abstract class Customer {

    Long id
    String info
    
    ...
}
```

Any idea why this is working?
